File: backend/models/invoice.py
from datetime import datetime
from pydantic import BaseModel
from enum import Enum

class InvoiceStatus(str, Enum):
    PENDING_UPLOAD = "PENDING_UPLOAD"
    PROCESSING = "PROCESSING"
    EXTRACTED = "EXTRACTED"
    FAILED = "FAILED"

class InvoiceMetadata(BaseModel):
    invoice_id: str
    s3_key: str
    filename: str
    content_type: str
    status: InvoiceStatus
    uploaded_at: datetime | None

File: frontend/types/Invoice.ts
export type InvoiceMetadata = {
    invoice_id: string;
    s3_key: string;
    content_type: string;
    filename: string;
    status: string;
    uploaded_at: string;
};

export type InvoicePage = {
    invoices:InvoiceMetadata[];
    next_cursor:string | null;
}